app/shared_channel_notifier.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+var SharedChannelEventsForSync model.StringArray = []string{
+	model.WEBSOCKET_EVENT_POSTED,
+	model.WEBSOCKET_EVENT_POST_EDITED,
+	model.WEBSOCKET_EVENT_POST_DELETED,
+	model.WEBSOCKET_EVENT_REACTION_ADDED,
+	model.WEBSOCKET_EVENT_REACTION_REMOVED,
+}
+
+// NotifySharedChannelSync signals the syncService to start syncing shared channel updates
+func (a *App) NotifySharedChannelSync(channel *model.Channel, event string) {
+	if !SharedChannelEventsForSync.Contains(event) || !channel.IsShared() {
```

Comment:
Minor optimization... since most channels are not shared, moving `!channel.IsShared()` to be the first expression will avoid walking the string array for most messages/reactions.